src/pages/GiveSkillNav.tsx

```diff
@@ -44,6 +49,24 @@ export const GiveSkillNav = ({ skill }: { skill: string }) => {
       >
         <TimelineList fa size="lg" /> Leaderboard
       </Button>
+      <Button
+        color="textOnly"
+        onClick={() => setModalVisible(true)}
```

Comment:
```suggestion
        onClick={() => setModalVisible(prev => !prev)}
```

Using a function updater (prev => !prev) in the setState function is a good practice when updating state that depends on the previous state value. This ensures that the state update is based on the most recent state, which is particularly important when state updates are batched or if there are multiple state updates in quick succession. @wingfeatherwave